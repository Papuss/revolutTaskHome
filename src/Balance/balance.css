.balance-component {
    margin-right: 10px;
}