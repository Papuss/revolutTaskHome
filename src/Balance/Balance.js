import "./balance.css";
import React from 'react';

function Balance(props) {
    return (
        <div className="balance-component" id={props.id}>
            <div className="balance">
            {"Balance: " + (props.currencyAccounts[props.currency] ? parseFloat(props.currencyAccounts[props.currency]).toFixed(2) : 0)}
            </div>
        </div>
    )
}

export default Balance;