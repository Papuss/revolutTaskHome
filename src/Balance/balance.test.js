import Balance from './Balance';
import enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';

enzyme.configure({ adapter: new Adapter() });

describe('Renders Balance component', () => {
it('should renders correctly', () => {
    const wrapper = shallow(
        <Balance 
            id="balance"
            currencyAccounts={{ "GBP": "2650", "USD": "1200", "EUR": "8500" }}
            currency={"HUF"}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
});
