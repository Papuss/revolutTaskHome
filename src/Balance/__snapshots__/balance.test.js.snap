// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Balance component should renders correctly 1`] = `ShallowWrapper {}`;
